impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -709,60 +718,73 @@ std::optional<Entity> GaussianBlurFilterContents::RenderFilter(
-      },
-      pass3_destination, blur_uvs);
-
-  if (!pass3_out.ok()) {
-    return std::nullopt;
-  }
-
   if (!renderer.GetContext()
            ->GetCommandQueue()
            ->Submit(/*buffers=*/{command_buffer})
            .ok()) {
     return std::nullopt;
   }
 
-  // The ping-pong approach requires that each render pass output has the same
-  // size.
-  FML_DCHECK((pass1_out.value().GetRenderTargetSize() ==
-              pass2_out.value().GetRenderTargetSize()) &&
-             (pass2_out.value().GetRenderTargetSize() ==
-              pass3_out.value().GetRenderTargetSize()));
-
   SamplerDescriptor sampler_desc = MakeSamplerDescriptor(
       MinMagFilter::kLinear, SamplerAddressMode::kClampToEdge);
 
-  Entity blur_output_entity = Entity::FromSnapshot(
-      Snapshot{.texture = pass3_out.value().GetRenderTargetTexture(),
-               .transform =
-                   entity.GetTransform() *                                   //
-                   Matrix::MakeScale(1.f / blur_info.source_space_scalar) *  //
-                   downsample_pass_args.transform *                          //
-                   Matrix::MakeScale(1 / downsample_pass_args.effective_scalar),
-               .sampler_descriptor = sampler_desc,
-               .opacity = input_snapshot->opacity},
-      entity.GetBlendMode());
-
-  return ApplyBlurStyle(mask_blur_style_, entity, inputs[0],
-                        input_snapshot.value(), std::move(blur_output_entity),
-                        mask_geometry_, blur_info.source_space_scalar);
+  Scalar y_sigma =
+      blur_info.scaled_sigma.y * downsample_pass_args.effective_scalar.y;
+  Matrix bar = Matrix::MakeScale(1.f / blur_info.source_space_scalar) *  //
+               downsample_pass_args.transform *                          //
+               Matrix::MakeScale(1 / downsample_pass_args.effective_scalar);
+  Matrix blur_output_entity_transform = entity.GetTransform() *  //
+                                        bar;
+  if (y_sigma < kEhCloseEnough) {
+    Entity blur_output_entity = Entity::FromSnapshot(
+        Snapshot{.texture = pass2_out.value().GetRenderTargetTexture(),
+                 .transform = blur_output_entity_transform,
+                 .sampler_descriptor = sampler_desc,
+                 .opacity = input_snapshot->opacity},
+        entity.GetBlendMode());
+    return ApplyBlurStyle(mask_blur_style_, entity, inputs[0],
+                          input_snapshot.value(), std::move(blur_output_entity),
+                          mask_geometry_, blur_info.source_space_scalar);
+  } else {
+    Entity blur_output_entity;
+    std::shared_ptr<Texture> texture =
+        pass2_out.value().GetRenderTargetTexture();
+    blur_output_entity.SetBlendMode(entity.GetBlendMode());
+    blur_output_entity.SetTransform(blur_output_entity_transform);
+    constexpr Vector2 orthographic_size(2.0f, 2.0f);
+    Vector2 inline_pixel_size = pass1_pixel_size / orthographic_size;
```

Comment:
@bdero @chinmaygarde I'm not sure if the is the correct math.  The 2.0 factor may be the dpi scalar and not like I was thinking an adjustment between the 2 coordinate systems.  Is there a better way to calculate the size of a pixel?

Reply:
Looking at the golden images (whose dpi scalar is 1.0) it really looks like that constant is the dpi scalar since the blurs are about half as big in the x direction.  We need another way to calculate the pixel size.

Reply:
Here is the same drawing on a 2.0x device and 3.0x device.  They look the same.  That makes it seem that the 2.0 denominator isn't related to DPI, it's just wrong.  It's probably slightly wrong since it looks good when eyeballing it.

<img width="901" alt="Screenshot 2024-07-01 at 10 11 13 AM" src="https://github.com/flutter/engine/assets/30870216/28517eaf-ad82-41e1-832c-b64cadf4f60b">

Reply:
I tried to calculate the pixel size by doing `entity.GetShaderTransform(pass).Transform(texture.size()) / texture.size()` and I get a pixel size that is about twice as big as what I was calculating which sounds correct given the golden results we saw.  That's also roughly the values we're using in the other blur pass (`1.0 / texture.size()`).

However I see banding.  It may be that the problem isn't the frequency of the blur sampling, but the phase of the sampling.  We are sampling at the right intervals, but at the wrong places.  The reason the 2.0 divisor has no banding is that we are hitting the nyquist frequency?

Reply:
I created a fudge slider that adjusts the uv sampling phase and that didn't effect the banding.